rosbag2_py/CMakeLists.txt

```diff
@@ -114,6 +123,11 @@ install(
   DESTINATION "${PYTHON_INSTALL_DIR}/${PROJECT_NAME}"
 )
 
+install(
+  TARGETS rosbag2_format_output
+  LIBRARY DESTINATION lib
+)
```

Comment:
@MichaelOrlov 
Yes.  
For Windows platform, it seems that `ARCHIVE` must be used in install().   
Could you run Windows CI again ?

Reply:
I can run CI one more time, but I am bit skeptical in success. Since the installation of the library going to happen after linker will try to find it for linkage to other targets. I think the `install` statement shall be right after the declaration of the `rosbag2_format_output` target and before the reference to it from other targets in the same CMake file.

Reply:
Re-run WIndows CI job
* Windows [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_windows&build=20687)](http://ci.ros2.org/job/ci_windows/20687/)

Reply:
> I can run CI one more time, but I am bit skeptical in success. Since the installation of the library going to happen after linker will try to find it for linkage to other targets. I think the install statement shall be right after the declaration of the rosbag2_format_output target and before the reference to it from other targets in the same CMake file.

My understanding is that due to dependency relationships, the library should already be installed. The placement of install() should not be relevant.

After investigation, I think it may be related to the differences in how libraries are handled on the Windows and Linux platforms.
Refer to CMake manual https://cmake.org/cmake/help/latest/command/install.html, `ARCHIVE` and `RUNTIME` parameters in install() are described as the below
![image](https://github.com/ros2/rosbag2/assets/43591679/d5091f39-8c25-4731-b287-78c98f0a0d4e)
![image](https://github.com/ros2/rosbag2/assets/43591679/88d8065a-6f73-4104-b8c0-7a455a5e8908)

According to failure log, compiler try to find `rosbag2_format_output.lib` (Not dll).  
So I think these 2 parameters are necessary for Windows platform.  
Sorry, I'm not familiar with building on the Windows platform. I cannot be entirely certain that making these changes will definitely resolve the issue.

Reply:
@MichaelOrlov 

According to failure log, the issue also exists.   
I think I have to prepare Windows environment to further check.

Reply:
@Barry-Xu-2018 Just a clue. Not sure if it will help for the current issue, but in a few cases it was helpful. 
It seems we are missing `ROSBAG2_PY_EXPORT` attribute in functions from the `rosbag2_format_output` library.

Reply:
@MichaelOrlov 

> It seems we are missing ROSBAG2_PY_EXPORT attribute in functions from the rosbag2_format_output library.

I searched the rosbag2 code, but I couldn't find the definition for ROSBAG2_PY_EXPORT.

Reply:
@Barry-Xu-2018 Yes. I meant we are missing the whole visibility control concept in the `rosbag2_py` C++ library. See example from another package https://github.com/ros2/rosbag2/blob/rolling/rosbag2_cpp/include/rosbag2_cpp/visibility_control.hpp

I guess for Python binding libraries it wasn't necessary. However, for C++ lib we need it.

Reply:
> I meant we are missing the whole visibility control concept in the `rosbag2_py` C++ library. See example from another package https://github.com/ros2/rosbag2/blob/rolling/rosbag2_cpp/include/rosbag2_cpp/visibility_control.hpp
> 
> I guess for Python binding libraries it wasn't necessary. However, for C++ lib we need it.

Get it. Thanks.  
I will try it. I am preparing window 10 environment.

Reply:
@MichaelOrlov 

Sorry. I spend much time to set up Window10 environment.   

> I guess for Python binding libraries it wasn't necessary. However, for C++ lib we need it.

You are right.  
I submitted [6543244](https://github.com/ros2/rosbag2/pull/1480/commits/6543244a6689f1c15316bc90b59e1840599e0a96). In this PR, I also adjusted the placement of the install function, grouping the definitions for the same target together.

Reply:
@Barry-Xu-2018 Thanks. :+1: I've restarted CI. Hopefully, it will pass now.